lib/live_view_google_maps_web/channels/maps.ex

```diff
@@ -0,0 +1,42 @@
+defmodule LiveViewGoogleMapsWeb.Maps do
+  require Logger
+
+  defstruct hash: nil
+
+  @moduledoc """
+  The Channel process for api audio.
+  """
+
+  @doc """
+    Subscribe to the topic/channel
+
+    ## Examples
+
+        iex> LiveViewGoogleMapsWeb.Maps.subscribe()
+        [_]
+  """
+  def subscribe(topic \\ "map") do
+    Phoenix.PubSub.subscribe(LiveViewGoogleMapsWeb.PubSub, topic)
+  end
+
+  @doc """
+  Push metrics to the topic/channel
+
+  ## Examples
+
+  iex> LiveViewGoogleMapsWeb.Maps.metrics()
+  [_]
+  """
+  def new_sighting(params \\ {:error, "metrics are empty"}) do
+    params
+    |> notify_subscribers([:new_sighting, :updated])
+  end
+
+  defp notify_subscribers({:ok, result}, event) do
+      Logger.info "notifying subscribers of : "<> @topic <> " " <> result
```

Comment:
did this work for you? I mean, I tried to run the code on my machine and i get the following error:
```
warning: undefined module attribute @topic, please remove access to @topic or explicitly set it before access
  lib/live_view_google_maps_web/channels/maps.ex:36: LiveViewGoogleMapsWeb.Maps (module)


== Compilation error in file lib/live_view_google_maps_web/channels/maps.ex ==
** (CompileError) lib/live_view_google_maps_web/channels/maps.ex:36: invalid literal nil in <<>>
    (elixir 1.10.3) src/elixir_bitstring.erl:34: :elixir_bitstring.expand/7
    (elixir 1.10.3) src/elixir_bitstring.erl:26: :elixir_bitstring.expand/4
    (elixir 1.10.3) expanding macro: Kernel.<>/2
    lib/live_view_google_maps_web/channels/maps.ex:36: LiveViewGoogleMapsWeb.Maps.notify_subscribers/2
    (logger 1.10.3) expanding macro: Logger.info/1
    lib/live_view_google_maps_web/channels/maps.ex:36: LiveViewGoogleMapsWeb.Maps.notify_subscribers/2
    (elixir 1.10.3) lib/kernel/parallel_compiler.ex:304: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

Reply:
was the intention to use the `topic` passed as parameter of the `def subscribe(topic \\ "map") do` method?